stl/inc/vector

```diff
@@ -1021,8 +1051,13 @@ private:
         const auto _Whereoff = static_cast<size_type>(_Where._Ptr - _Myfirst);
         const auto _Oldsize  = static_cast<size_type>(_Mylast - _Myfirst);
 
+        // Pessimistically assume that we will fill the vector's capacity.
+        // This causes the entire underlying buffer to initially be marked as legal.
+        _ASAN_ANNOTATE(*this, static_cast<size_type>(_My_data._Myend - _Myfirst) + 1);

```

Comment:
I want the annotator to assume that the underlying buffer is going to resize so that it marks [last, end) as accessible.

Reply:
Ah - I think I get it. Since this _might_ allocate a new memory block, it needs to use the "new memory block" strategy. (This is another reason why "new memory block" must allow access to the entire old block.) `capacity() + 1` here is a convenient number that we _know_ will be greater than capacity, and not the copy-pasta it appears to be. Do you think it might be clearer to use something enormous here like `~size_type{0}` which is (1) clearly not copy-pasta, and (2) possibly avoids unnecessary calculation?